### PR TITLE
planner/core: give the full schema information when group by fields not exists

### PR DESCRIPTION
**Pull request by gaoxingliang**
### What problem does this PR solve? 
Closes https://github.com/pingcap/tidb/issues/11145
the error message didn't contains a valid column name when group by failed.

### What is changed and how it works?
when generating the error message compose it to dbName.tableName.colName format

### Check List 

Tests 

 - Unit test   - Passed and Added 2 new test cases

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -1768,7 +1768,14 @@ func (b *PlanBuilder) checkOnlyFullGroupByWithGroupClause(p LogicalPlan, sel *as
 		}
 		switch errExprLoc.Loc {
 		case ErrExprInSelect:
-			return ErrFieldNotInGroupBy.GenWithStackByArgs(errExprLoc.Offset+1, errExprLoc.Loc, sel.Fields.Fields[errExprLoc.Offset].Text())
+			var errText string
+			selectField := sel.Fields.Fields[errExprLoc.Offset]
+			if colExpr, ok := selectField.Expr.(*ast.ColumnNameExpr); ok && len(schema.Columns) > 0 {
```
Comment (SunRunAway) [inline planner/core/logical_plan_builder.go]: Please consider other possible expressions like `Length(field)`

File: planner/core/logical_plan_builder.go [hunk 2]
```diff
@@ -1768,7 +1768,14 @@ func (b *PlanBuilder) checkOnlyFullGroupByWithGroupClause(p LogicalPlan, sel *as
...
+				errText = fmt.Sprintf("%s.%s.%s", schema.Columns[0].DBName.L, tblInfo.Name.L, colExpr.Name.Name.L)
```
Comment (SunRunAway) [inline planner/core/logical_plan_builder.go]: If the table has an alias...

File: planner/core/logical_plan_builder.go (hunk 1 above)
Comment (gaoxingliang) [reply planner/core/logical_plan_builder.go]: Yes. and fixed by use a much more easier solution. and UT added

File: planner/core/logical_plan_builder.go (hunk 2 above)
Comment (gaoxingliang) [reply planner/core/logical_plan_builder.go]: Yes. and fixed by use a much more easier solution. and UT added

File: planner/core/logical_plan_builder.go [hunk 3]
```diff
@@ -1768,7 +1768,8 @@ func (b *PlanBuilder) checkOnlyFullGroupByWithGroupClause(p LogicalPlan, sel *as
 		}
 		switch errExprLoc.Loc {
 		case ErrExprInSelect:
-			return ErrFieldNotInGroupBy.GenWithStackByArgs(errExprLoc.Offset+1, errExprLoc.Loc, sel.Fields.Fields[errExprLoc.Offset].Text())
+			errText := fmt.Sprintf("%s.%s.%s", col.DBName.O, col.OrigTblName.O, col.OrigColName.O)
```
Comment (SunRunAway) [inline planner/core/logical_plan_builder.go]: I tested in mysql5 and mysql8, they both use table alias in the error message.

File: planner/core/logical_plan_test.go [hunk 4]
```diff
@@ -15,6 +15,7 @@ package core
 
 import (
 	"fmt"
+	"github.com/pingcap/parser/ast"
```
Comment (SunRunAway) [inline planner/core/logical_plan_test.go]: Please use [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports) to sort the imports.

File: planner/core/logical_plan_builder.go (hunk 3 above)
Comment (lzmhhh123) [inline planner/core/logical_plan_builder.go]: Please remove the `fmt.Sprint`. It's inefficient. Just plus the string and no need to declare a new variable either.

File: planner/core/logical_plan_test.go (hunk 4 above)
Comment (gaoxingliang) [reply planner/core/logical_plan_test.go]: fixed

File: planner/core/logical_plan_builder.go (hunk 3 above)
Comment (gaoxingliang) [reply planner/core/logical_plan_builder.go]: fixed

File: planner/core/logical_plan_test.go [hunk 5]
```diff
@@ -19,6 +19,8 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/pingcap/parser/ast"
+
```
Comment (zz-jason) [inline planner/core/logical_plan_test.go]: please remove this empty line.

Comment (gaoxingliang) [reply planner/core/logical_plan_test.go]: fixed. I assumed all empty lines should be removed right?

Comment (zz-jason) [reply planner/core/logical_plan_test.go]: No. We split the import packages to two parts. The first part is the golang standard libs, the second part is the third-part libs. Between these two parts, there should be an empty line.

Comment (gaoxingliang) [reply planner/core/logical_plan_test.go]: Aha. Got it. Thanks

File: planner/core/logical_plan_test.go [hunk 6]
```diff
@@ -561,6 +562,71 @@ func (s *testPlanSuite) TestDeriveNotNullConds(c *C) {
...
+	return BuildLogicalPlan(s.ctx, stmt, s.is)
+}
+
+func (s *testPlanSuite) TestGroupByWhenNotExistCols(c *C) {
+	sqlTests := []struct {
+		sql              string
+		expectedErrMatch string
+	}{
+		{
```
Comment (winoros) [inline planner/core/logical_plan_test.go]: Add a case that the table is aliased twice and column is aliased twice too.

Comment (gaoxingliang) [reply planner/core/logical_plan_test.go]: do you mean a sql like:
```
select age from (select id as x, age from ss as tableAlias) as tableAlias2 group by tableAlias2.x;
```
with below schema:
```
mysql> desc ss;
+-------+---------+------+-----+---------+-------+
| Field | Type    | Null | Key | Default | Extra |
+-------+---------+------+-----+---------+-------+
| id    | int(11) | YES  |     | NULL    |       |
| age   | int(11) | YES  |     | NULL    |       |
| task  | int(11) | YES  |     | NULL    |       |
+-------+---------+------+-----+---------+-------+
3 rows in set (0.01 sec)

```

Comment (winoros) [reply planner/core/logical_plan_test.go]: okay

**Comment by SunRunAway** [comment]
@gaoxingliang please take a look at the CI failure.

**Comment by gaoxingliang** [comment]
Merge from master to make compile work. and it seems the CLA is blocking? the committer is changed.

**Comment by SunRunAway** [comment]
> edward.gao seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please add the email address used for this commit to your account.

Hi, it seems you should set your commit email address in Git.

**Comment by gaoxingliang** [comment]
@SunRunAway I have changed the committer for edward.gao (renamed to gaoxingliang). but now it seems blocked.

**Comment by SunRunAway** [comment]
It seems you need to recreate all the commit message. Try
```
git fetch upstream master
git rebase upstream/master
git reset upstream/master
git commit
git push -f
```

**Comment by gaoxingliang** [comment]
Thanks @SunRunAway  it worked.

**Comment by SunRunAway** [comment]
@gaoxingliang great, could you update your checklist in the pull request description, remove the lines which are not suitable.

**Comment by gaoxingliang** [comment]
it seems failed because of tikv errors?
```
[2019-07-25T09:26:45.094Z] OOPS: 98 passed, 1 FAILED

[2019-07-25T09:26:45.094Z] --- FAIL: TestT (46.92s)

[2019-07-25T09:26:45.094Z] FAIL

[2019-07-25T09:26:45.094Z] FAIL	github.com/pingcap/tidb/store/tikv	47.887s

```

**Comment by gaoxingliang** [comment]
Another case failed:

> [2019-07-25T10:59:31.057Z] FAIL: executor_test.go:4335: testRecoverTable.TestRecoverTable
[2019-07-25T10:59:31.057Z] 
[2019-07-25T10:59:31.057Z] executor_test.go:4408:
[2019-07-25T10:59:31.057Z]     // check rebase auto id.
[2019-07-25T10:59:31.057Z]     tk.MustQuery("select a,_tidb_rowid from t_recover;").Check(testkit.Rows("1 1", "2 2", "3 3", "4 5001", "5 5002", "6 5003"))
[2019-07-25T10:59:31.057Z] /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:61:
[2019-07-25T10:59:31.057Z]     res.c.Assert(resBuff.String(), check.Equals, needBuff.String(), res.comment)
[2019-07-25T10:59:31.057Z] ... obtained string = "" +
[2019-07-25T10:59:31.057Z] ...     "[1 4]\n" +
[2019-07-25T10:59:31.057Z] ...     "[2 5]\n" +
[2019-07-25T10:59:31.057Z] ...     "[3 6]\n" +
[2019-07-25T10:59:31.057Z] ...     "[4 5001]\n" +
[2019-07-25T10:59:31.057Z] ...     "[5 5002]\n" +
[2019-07-25T10:59:31.057Z] ...     "[6 5003]\n"
[2019-07-25T10:59:31.057Z] ... expected string = "" +
[2019-07-25T10:59:31.057Z] ...     "[1 1]\n" +
[2019-07-25T10:59:31.057Z] ...     "[2 2]\n" +
[2019-07-25T10:59:31.057Z] ...     "[3 3]\n" +
[2019-07-25T10:59:31.057Z] ...     "[4 5001]\n" +
[2019-07-25T10:59:31.057Z] ...     "[5 5002]\n" +
[2019-07-25T10:59:31.057Z] ...     "[6 5003]\n"
[2019-07-25T10:59:31.057Z] ... sql:select a,_tidb_rowid from t_recover;, args:[]

**Pull request merged**